SCons/Scanner/C.py

```diff
@@ -61,15 +61,20 @@ def read_file(self, file):
             self.missing.append((file, self.current_file))
             return ''
 
-def dictify_CPPDEFINES(env):
+def dictify_CPPDEFINES(env) -> dict:
+    """Returns CPPDEFINES converted to a dict."""
     cppdefines = env.get('CPPDEFINES', {})
     if cppdefines is None:
         return {}
     if SCons.Util.is_Sequence(cppdefines):
         result = {}
         for c in cppdefines:
             if SCons.Util.is_Sequence(c):
-                result[c[0]] = c[1]
+                try:
+                    result[c[0]] = c[1]
+                except IndexError:
```

Comment:
Honest question: Is it how we roll in python? Just catch exception? I was thinking about guarding it with something like `if size(c) == 0`

Reply:
There's no one fixed answer... the Python coredevs are promising the next release will have "zero-cost" exceptions, there's always a small cost to either EAFP (easier to ask forgiveness than permission) or LBYL (look before you leap), so sometimes the choice comes down to which case is more likely if it's performance-seneitive.  A lot of Python devs are in the EAFP camp, though.

Reply:
I've done some profiling (though not with 3.10) of such. It really comes down to what's the most common occurence. If we expect that we'll have lots of single element tuples in the list, then LBYL is a better choice, if not then EAFP.  At least as of 3.7 or 3.8 when last I line_profile'd parts of subst logic..

Reply:
Thank you for an extensive answer!